vyper/codegen/module.py

```diff
@@ -100,6 +100,12 @@ def _runtime_ir(runtime_functions, all_sigs, global_ctx):
     internal_functions = [f for f in runtime_functions if _is_internal(f)]
 
     external_functions = [f for f in runtime_functions if not _is_internal(f)]
+
+    # check if any selector is 0
+    has_zero_selector_functions = any(
+        [0 in ContractFunction.from_FunctionDef(f).method_ids.values() for f in external_functions]
```

Comment:
you don't need to call `ContractFunction.from_FunctionDef`. it's already on the function object -- see other uses like `func_ast._metadata["type"]`